tests/conformance/scripts/run-tests/index.ts

```diff
@@ -41,51 +162,13 @@ async function main(options: {
   let hadError = false;
   let foundAnyTests = false;
   for (const runner of enabledRunners) {
-    console.log("running", runner.name);
-    for (const variant of generateTestVariants()) {
-      if (options.testRegex && !options.testRegex.test(variant.name)) {
-        continue;
-      }
-      foundAnyTests = true;
-      const filePath = path.join(options.dataDir, variant.baseName, `${variant.name}.mcap`);
-
-      if (runner.supportsVariant(variant)) {
-        console.log("  testing", filePath);
-      } else {
-        console.log("  not supported", filePath);
-        continue;
-      }
-
-      let output: string;
-      try {
-        output = await runner.run(filePath, variant);
-      } catch (error) {
-        console.error(error);
-        hadError = true;
-        continue;
-      }
-      const expectedOutputPath = filePath.replace(/\.mcap$/, ".json");
-      if (options.update) {
-        await fs.writeFile(expectedOutputPath, output);
-      } else {
-        const expectedOutput = await fs
-          .readFile(expectedOutputPath, { encoding: "utf-8" })
-          .catch(() => undefined);
-        if (expectedOutput == undefined) {
-          console.error(`Error: missing expected output file ${expectedOutputPath}`);
-          hadError = true;
-          continue;
-        }
-        const outputNorm = normalizeJson(output);
-        const expectedNorm = normalizeJson(expectedOutput);
-        if (outputNorm !== expectedNorm) {
-          console.error(`Error: output did not match expected for ${filePath}:`);
-          console.error(Diff.createPatch(expectedOutputPath, expectedNorm, outputNorm));
-          hadError = true;
-          continue;
-        }
-      }
-    }
+    const testFunction = runner.mode === "read" ? runReaderTest : runWriterTest;
+    const { hadError: newHadError, foundAnyTests: newFoundAnyTests } = await testFunction(
+      runner,
+      options,
+    );
+    hadError ||= newHadError;
+    foundAnyTests ||= newFoundAnyTests;
```

Comment:
`foundAnyTests` is going to be the same for all readers/writers test runs. It might make more sense to just move it out of the runReaderTest/runWriterTest functions.